components/chat/chat-container.tsx

```diff
@@ -1,140 +1,127 @@
-'use client';
-import React, { useContext, useEffect, useState } from 'react';
-import { Card, Grid, Skeleton, AspectRatio, Box, aspectRatioClasses } from '@/lib/mui';
-import { useRequest } from 'ahooks';
-import useAgentChat from '@/hooks/use-agent-chat';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { useAsyncEffect } from 'ahooks';
+import useChat from '@/hooks/use-chat';
 import Completion from '@/components/chat/completion';
-import { ChartData } from '@/types/chart';
-import { apiInterceptors, getChatHistory, postChatModeParamsList } from '@/client/api';
+import { ChartData, ChatHistoryResponse } from '@/types/chart';
+import { apiInterceptors, getChatHistory } from '@/client/api';
 import { ChatContext } from '@/app/chat-context';
-import MuiLoading from '../common/loading';
 import Header from './header';
 import Chart from '../chart';
 import classNames from 'classnames';
-
-const ChartSkeleton = () => {
-  return (
-    <Card className="h-full w-full flex bg-transparent">
-      <Skeleton animation="wave" variant="text" level="body2" />
-      <Skeleton animation="wave" variant="text" level="body2" />
-      <AspectRatio
-        ratio="21/9"
-        className="flex-1"
-        sx={{
-          [`& .${aspectRatioClasses.content}`]: {
-            height: '100%',
-          },
-        }}
-      >
-        <Skeleton variant="overlay" className="h-full" />
-      </AspectRatio>
-    </Card>
-  );
-};
+import MuiLoading from '../common/loading';
+import { Empty } from 'antd';
+import { useSearchParams } from 'next/navigation';
 
 const ChatContainer = () => {
+  const searchParams = useSearchParams();
+  const { scene, chatId, model, setModel } = useContext(ChatContext);
+  const chat = useChat({});
+  const initMessage = (searchParams && searchParams.get('initMessage')) ?? '';
```

Comment:
`initMessage` in url is not a good ideal, maybe `PostMessage` or `Broadcast Channel`?